src/utils/assignable.js

```diff
@@ -3,43 +3,63 @@
 // Do not use any browser or node-specific API!
 // -------------------------------------------------------------
 
+import { ActionInvalidObjectPropertyError } from '../shared/errors';
+
+function validateObjectProps (obj, dest) {
+    const objectName = dest.constructor.name;
+    const validKeys = dest._getAssignableProperties().map(p => p.name);
+    const reportedProperties = dest._getReportedProperties();
+
+    for (const key in obj) {
+        if (!(validKeys.includes(key) || key in dest))
+            throw new ActionInvalidObjectPropertyError(objectName, key, reportedProperties);
+    }
```

Comment:
This code is used on the client and server sides. 
For such code files we added a specific comment at the top of file:
```js
// Do not use any browser or node-specific API!
```
The tested web site can override the `Array.includes` method and in this case TestCafe will be broken. You need to rewrite this using the raw operators or store used methods at the top the file as follows:
```js
const arrayIncludes = Array.prototype.includes;

function validateObjectProps (obj, dest) {
...
```